dropwizard-json-logging/src/main/java/io/dropwizard/logging/json/layout/AccessJsonLayout.java

```diff
@@ -0,0 +1,116 @@
+        super(jsonFormatter);
+        this.timestampFormatter = timestampFormatter;
+        this.additionalFields = additionalFields;
+        this.customFieldNames = customFieldNames;
+    }
+
+    @Override
+    protected Map<String, Object> toJsonMap(IAccessEvent event) {
+        return new MapBuilder(timestampFormatter, customFieldNames, additionalFields, 20)
+            .add("port", isIncluded(AccessAttribute.LOCAL_PORT), event.getLocalPort())
+            .add("contentLength", isIncluded(AccessAttribute.CONTENT_LENGTH), event.getContentLength())
+            .addTimestamp("timestamp", isIncluded(AccessAttribute.TIMESTAMP), event.getTimeStamp())
+            .add("method", isIncluded(AccessAttribute.METHOD), event.getMethod())
+            .add("protocol", isIncluded(AccessAttribute.PROTOCOL), event.getProtocol())
+            .add("requestContent", isIncluded(AccessAttribute.REQUEST_CONTENT), event.getRequestContent())
+            .add("remoteAddress", isIncluded(AccessAttribute.REMOTE_ADDRESS), event.getRemoteAddr())
+            .add("remoteUser", isIncluded(AccessAttribute.REMOTE_USER), event.getRemoteUser())
+            .add("headers", !requestHeaders.isEmpty(),
+                filterHeaders(event.getRequestHeaderMap(), requestHeaders))
+            .add("params", isIncluded(AccessAttribute.REQUEST_PARAMETERS), event.getRequestParameterMap())
+            .add("requestTime", isIncluded(AccessAttribute.REQUEST_TIME), event.getElapsedTime())
+            .add("uri", isIncluded(AccessAttribute.REQUEST_URI), event.getRequestURI())
+            .add("url", isIncluded(AccessAttribute.REQUEST_URL), event.getRequestURL())
+            .add("remoteHost", isIncluded(AccessAttribute.REMOTE_HOST), event.getRemoteHost())
+            .add("responseContent", isIncluded(AccessAttribute.RESPONSE_CONTENT), event.getResponseContent())
+            .add("responseHeaders", !responseHeaders.isEmpty(),
+                filterHeaders(event.getResponseHeaderMap(), responseHeaders))
+            .add("serverName", isIncluded(AccessAttribute.SERVER_NAME), event.getServerName())
+            .add("status", isIncluded(AccessAttribute.STATUS_CODE), event.getStatusCode())
+            .add("userAgent", isIncluded(AccessAttribute.USER_AGENT), event.getRequestHeader(HttpHeaders.USER_AGENT))
+            .add("version", jsonProtocolVersion != null, jsonProtocolVersion)
+            .build();
+    }
+
+    private boolean isIncluded(AccessAttribute userAgent) {
+        return includes.contains(userAgent);
+    }
+
+    private Map<String, String> filterHeaders(Map<String, String> headers, Set<String> filteredHeaderNames) {
+        if (filteredHeaderNames.isEmpty()) {
+            return ImmutableMap.of();
+        }
+        return headers.entrySet().stream()
+            .filter(e -> filteredHeaderNames.stream().anyMatch(h -> h.equalsIgnoreCase(e.getKey())))
```

Comment:
Can the filteredHeaderNames be put into a case insensitive TreeSet? This this nested loop could be converted to a contains.

Reply:
Good idea! I will use `ImmutableSortedSet` from Guava for that (the same behavior, but it's a contagious set and provides a better memory layout than `TreeSet`).